client/src/app/modules/events/event-edit/event-edit.component.ts

```diff
@@ -1,66 +1,169 @@
+    this.iEvent.endDate = this.fromBrowserDate(this.eventForm.controls.dateEnd.value);
+    this.iEvent.description = this.eventForm.controls.description.value;
+    this.iEvent.limit = this.eventForm.controls.participantsLimit.value;
+    this.iEvent.transportBus = toBoolean(this.eventForm.controls.transport.value.bus);
+    this.iEvent.transportCar = toBoolean(this.eventForm.controls.transport.value.car);
+    this.iEvent.accommodation = toBoolean(this.eventForm.controls.accommodation.value);
+  }
+
+  /**
+   * Structure definition needed for template rendering
+   */
+  private getEventForm(): FormGroup {
+    return new FormGroup({
+      'name': new FormControl(null),
+      'location': new FormControl(null),
+      'dateStart': new FormControl(null),
+      'dateEnd': new FormControl(null),
+      'description': new FormControl(null),
+      'participantsLimit': new FormControl(null),
+      'transport': new FormGroup({
+        'bus': new FormControl(null),
+        'car': new FormControl(null)
+      }),
+      'accommodation': new FormControl(null)
+    });
+  }
+
+  private setEventFormValues(iEvent: IEventSchema): void {
+    const todayString = this.toBrowserDate(new Date());
+    const valueOrDefault = (value: any, defaultValue?: any): any => value ? value : defaultValue ? defaultValue : '';
+    this.eventForm.setValue({
+      'name': valueOrDefault(iEvent.name),
+      'location': iEvent.location ? valueOrDefault(iEvent.location.address) : '',
+      'dateStart': valueOrDefault(this.toBrowserDate(iEvent.startDate), todayString),
+      'dateEnd': valueOrDefault(this.toBrowserDate(iEvent.endDate), todayString),
+      'description': valueOrDefault(iEvent.description),
+      'participantsLimit': valueOrDefault(iEvent.limit),
+      transport: {
+        'car': valueOrDefault(iEvent.transportBus),
+        'bus': valueOrDefault(iEvent.transportCar)
+      },
+      'accommodation': valueOrDefault(iEvent.accommodation.toString())
+    });
+  }
+
+  private setEventFormValidators(): void {
+    this.eventForm.controls.name.setValidators([Validators.required, Validators.minLength(5), Validators.maxLength(140)]);
+    this.eventForm.controls.location.setValidators([Validators.required]);
+    this.eventForm.controls.dateStart.setValidators([Validators.required]);
+    this.eventForm.controls.dateEnd.setValidators([Validators.required]);
+    this.eventForm.controls.description.setValidators([]);
+    this.eventForm.controls.participantsLimit.setValidators([Validators.pattern('\\s*[1-9]+[0-9]*\\s*')]);
+    (<FormGroup>this.eventForm.controls['transport']).controls['car'].setValidators([]);
+    (<FormGroup>this.eventForm.controls['transport']).controls['bus'].setValidators([]);
+    this.eventForm.controls.accommodation.setValidators([]);
+  }
+
+  private toBrowserDate(date: Date): String {
+    const properDate = new Date(date.toString()); // Dirty hack, because sometimes date is passed as String, like WTF TypeScript?!
```

Comment:
This is not WTF typescript, this is **WTF javascript**! 
I had the same issue on our project... there are several ways how to fix this JS shit :) 
My preferred way (which was not preferred way of my colleagues) was to create request interceptor that will always check the whole incoming JSON and if it finds date in string ISO format it will be automatically parsed as Date (new Date('timestamp')). There are already working (copy&paste) solutions on the web.